### PR TITLE
feat(xbind): Add support for x:Bind to POCO, performance improvements

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes https://github.com/unoplatform/uno/issues/3240, Fixes https://github.com/unoplatform/uno/issues/3598, Fixes https://github.com/unoplatform/uno/issues/3597

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

This change adds support for:
- x:Bind to POCO
- `Binding.Update()`
- Removes tree traversal for x:Bind and resources updates

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by carldebilly — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs [hunk 1]
```diff
@@ -555,7 +555,6 @@ public void RestoreBindings()
 
 					public void ApplyCompiledBindings()
 					{{
```
Comment (davidjohnoliver) [inline src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:557]: maybe a comment here (eg 'this is left for binary compatibility but...')

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 2]
```diff
@@ -1448,6 +1532,16 @@ private bool HasBindingMarkupExtension(XamlMemberDefinition valueNode)
...
 
+		private bool HasXBindMarkupExtension(XamlObjectDefinition objectDefinition)
+			=> objectDefinition
+				.Members
+				.Any(o => o.Objects.Any(o => o.Type.Name == "Bind"));
+
+		private bool HasResourceMarkupExtension(XamlObjectDefinition objectDefinition)
+			=> objectDefinition
+				.Members
+				.Any(o => o.Objects.Any(o => o.Type.Name == "Bind" || o.Type.Name == "StaticResource" || o.Type.Name == "ThemeResource"));
```
Comment (davidjohnoliver) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:1543]: Is the `o.Type.Name == "Bind"` intended here?

**Review by davidjohnoliver — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs (hunk 2 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:1543]: Yes it is. It's supposed to be used for "component" backing field generation, but the name of the method is not appropriate. I'll update the name.

File: src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs:557]: Will do in next PR.

**Pull request merged**